### PR TITLE
MSA label update

### PR DESCRIPTION
**Pull request by alexdinug**

### CODE REVIEW

**Review by discodavey — CHANGES_REQUESTED** [review]
All attributes in the classes need to be in alphbetical order. Also can you use the mixins for padding and margin where possible

File: assets/sass/patterns/organisms/content-header.scss [hunk 1]
```diff
@@ -414,6 +399,10 @@
   }
 }
 
+.content-header__subject_link:hover {
+  text-decoration: none;
```
Comment (discodavey) [inline assets/sass/patterns/organisms/content-header.scss:403]: This can be included in &:hover on 390

File: assets/sass/patterns/organisms/content-header-journal.scss [hunk 2]
```diff
@@ -38,6 +38,30 @@
   }
 }
 
+.content-header__subject_link {
+  text-decoration: none;
+
+  .content-header__subject {
+    @include margin(12, "right");
+    @include padding(5 10 3 10);
+    background-color: #ebf2fa;
```
Comment (discodavey) [inline assets/sass/patterns/organisms/content-header-journal.scss:47]: Can we check these colours please as they do not appear in variables.scss

File: assets/sass/patterns/organisms/content-header-journal.scss [hunk 3]
```diff
@@ -38,6 +38,30 @@
...
+    border-radius: 3px;
+    color: $color-primary;
+    font-weight: bold;
+  }
+}
+
+.content-header__subject_list_item > .content-header__subject {
+  @include font-size-and-vertical-height(12);
+  @include margin(12, "right");
+  background-color: #edeff4;
```
Comment (discodavey) [inline assets/sass/patterns/organisms/content-header-journal.scss:57]: Can we check these colours with the design and Chris please as they do not appear in variables.scss

File: assets/sass/patterns/organisms/content-header-journal.scss [hunk 4]
```diff
@@ -38,6 +38,30 @@
...
+    @include margin(12, "right");
+    @include padding(5 10 3 10);
+    background-color: #ebf2fa;
+    border-radius: 3px;
+    color: $color-primary;
+    font-weight: bold;
+  }
+}
+
+.content-header__subject_list_item > .content-header__subject {
```
Comment (discodavey) [inline assets/sass/patterns/organisms/content-header-journal.scss:54]: Can we break this down so there is no specifity if possible?

File: assets/sass/patterns/molecules/teaser.scss [hunk 5]
```diff
@@ -25,47 +25,50 @@
...
 
-.teaser__context_label_link:after {
-  content: ", ";
+.teaser .teaser__context_label_link:hover {
+  text-decoration: none;
 }
 
-.teaser__context_label_link:last-child:after {
-  content: "";
+.article-secondary-column .teaser .teaser__context_label {
```
Comment (discodavey) [inline assets/sass/patterns/molecules/teaser.scss:70]: Can we break this down if possible as we do not want specificity if it cannot be helped.

**Review by discodavey — CHANGES_REQUESTED** [review]
Whitespace needs removing.

Screenshot 2022-08-22 at 09 13 37

File: assets/sass/patterns/molecules/teaser.scss [hunk 6]
```diff
@@ -25,47 +25,46 @@
...
-
 }
 
 .teaser__context_label {
   // content-label rather than subject-label as not it's not always subjects in here
-  @include label-content-typeg();
+  @include font-size-and-vertical-height(12);
+  @include margin(12, "right");
+  @include padding(5 10 3 10);
+  background-color: #edeff4;
```
Comment (discodavey) [inline assets/sass/patterns/molecules/teaser.scss:38]: Can be shown as a variable that is available

File: assets/sass/patterns/molecules/teaser.scss [hunk 7]
```diff
@@ -25,47 +25,46 @@
...
 .article-secondary-column .teaser .teaser__context_label {
-  color: $color-text-secondary;
-}
-
-.teaser__context_label_link:after {
-  content: ", ";
+  @include font-size-and-vertical-height(12);
+  @include margin(12, "right");
+  @include padding(5 10);
+  background-color: #edeff4;
```
Comment (discodavey) [inline assets/sass/patterns/molecules/teaser.scss:59]: Can be shown as a variable that is available

File: source/_patterns/01-molecules/teasers/teaser.mustache [hunk 8]
```diff
@@ -9,7 +9,6 @@
           {{#url}}
             <a href="{{url}}"  {{#carouselItem}}tabindex="-1"{{/carouselItem}} class="teaser__context_label_link"><span class="teaser__context_label">{{{name}}}</span></a>
           {{/url}}
-
```
Comment (nlisgo) [inline source/_patterns/01-molecules/teasers/teaser.mustache:12]: This could be undone.

File: source/_patterns/02-organisms/content-headers/content-header-journal.mustache [hunk 9]
```diff
@@ -15,9 +15,14 @@
...
-                  <span class="content-header__subject">{{{name}}}</span>
-                </a>
+                  {{#url}}
+                      <a href="{{url}}" class="content-header__subject_link">
+                          <span class="content-header__subject">{{{name}}}</span>
+                      </a>
+                  {{/url}}
+                  {{^url}}
+                      <span class="content-header__subject">{{{name}}}</span>
+                  {{/url}}
```
Comment (nlisgo) [inline source/_patterns/02-organisms/content-headers/content-header-journal.mustache:25]: Undo this change because url is required for msa

File: source/_patterns/04-pages/article--research.json [hunk 10]
```diff
@@ -150,8 +150,20 @@
...
+          "name": "Epidemiology and Global Health",
+          "url": ""
+        },
+        {
+          "name": "Epidemiology and Global Health",
+          "url": ""
+        },
+        {
+          "name": "Epidemiology and Global Health",
+          "url": ""
```
Comment (nlisgo) [inline source/_patterns/04-pages/article--research.json:166]: Would be better to create a variant on a content-header rather than a page.

File: source/_patterns/02-organisms/content-headers/content-header-journal.mustache (hunk 9 above)
Comment (nlisgo) [inline source/_patterns/02-organisms/content-headers/content-header-journal.mustache:25]: Undo this change because url is required for msa

**Review by nlisgo — CHANGES_REQUESTED** [review]

**Review by discodavey — CHANGES_REQUESTED** [review]
Add 12px to bottom of breadcrumbs so there is correct spacing between the breadcrumb and MSA. Make sure this does not affect any other part of the content headers.

Screenshot 2022-08-31 at 11 09 01

**Review by discodavey — CHANGES_REQUESTED** [review]
Spacing between each of the MSA's is too wide. Should be 6px inbetween each MSA.

Screenshot 2022-08-31 at 11 16 47

**Comment by discodavey** [comment]
@chris-huggins Is this background colour for "Built upon by" correct?

Screenshot 2022-08-31 at 11 43 40

**Comment by discodavey** [comment]
@alexdinug I have added some new feedback for the MSA's and there are some outstanding issues that need to be completed. Can you let me know when this has been done so we can then get this PR merged.

**Comment by chris-huggins** [comment]
> @chris-huggins Is this background colour for "Built upon by" correct?
> 
> Screenshot 2022-08-31 at 11 43 40

I hadn't intended for these labels to have a background. I'm not sure how it has happened, but would like to keep the existing design for "built upon by", "builds upon", "related to" etc. labels for related articles.

**Comment by Cosmin-Daniel** [comment]
> > @chris-huggins Is this background colour for "Built upon by" correct?
> > Screenshot 2022-08-31 at 11 43 40
> 
> I hadn't intended for these labels to have a background. I'm not sure how it has happened, but would like to keep the existing design for "built upon by", "builds upon", "related to" etc. labels for related articles.

Aren't these context (MSA) labels? In the code seem to have same classes. Do we have a specific design for the secondary teasers (teasers in the second column)? I remember I asked about those ones on the homepage and you confirmed they should have background too. Please let me know if I need to change just the teaser list on the main column (homepage and article page). I am also going to do the changes on a teaser variant, as Nathan requested.

**Comment by chris-huggins** [comment]
@Cosmin-Daniel No sorry that screenshot shows a different type of link that is not an MSA. You're correct that I was ok with using the background on MSA labels in the right column/teasers on homepage, MSA pages etc.. But these labels appear on article pages, so I'm guessing, as you say, it's because they have the same class.

I think rather than trying to change this, let's just revert/leave the styling of MSA labels in the right column/teasers.
Therefore to answer this...
> Please let me know if I need to change just the teaser list on the main column (homepage and article page).

Yes let's just style the teasers on the main columns on homepage and MSA listings please. But not articles.

I would like to delay changing the MSA labels on article pages. Or at least delay deploying them to journal until we make other changes to the article pages. At the moment they conflict with the article type (e.g. Research article).

image

**Comment by nlisgo** [comment]
@chris-huggins 

> I would like to delay changing the MSA labels on article pages. Or at least delay deploying them to journal until we make other changes to the article pages.

Are you saying that we should only be amend the msa labels in the main teasers as part of this ticket?

**Review by discodavey — APPROVED** [review]
I am happy with this if @chris-huggins is. @nlisgo Have you got any other comments?

**Comment by Cosmin-Daniel** [comment]
> @Cosmin-Daniel No sorry that screenshot shows a different type of link that is not an MSA. You're correct that I was ok with using the background on MSA labels in the right column/teasers on homepage, MSA pages etc.. But these labels appear on article pages, so I'm guessing, as you say, it's because they have the same class.
> 
> I think rather than trying to change this, let's just revert/leave the styling of MSA labels in the right column/teasers. Therefore to answer this...
> 
> > Please let me know if I need to change just the teaser list on the main column (homepage and article page).
> 
> Yes let's just style the teasers on the main columns on homepage and MSA listings please. But not articles.
> 
> I would like to delay changing the MSA labels on article pages. Or at least delay deploying them to journal until we make other changes to the article pages. At the moment they conflict with the article type (e.g. Research article).
> 
> image

Ok. The changes are done. I reverted the changes on the article page and changed the style in order to have modified just the homepage main column and the MSA listing pages. Thank you for the clarifications!

File: assets/sass/patterns/molecules/teaser.scss [hunk 11]
```diff
@@ -78,6 +78,33 @@
...
+    @include margin(12, "bottom");
+    @include margin(6, "right");
+    display: inline-block;
+
+    & .teaser__context_label:after {
+      content: "";
+    }
+  }
+  .teaser__context_label {
+    // content-label rather than subject-label as not it's not always subjects in here
```
[the remark this answers is not in the document]
Comment (Cosmin-Daniel) [reply assets/sass/patterns/molecules/teaser.scss]: The text is corrected now

**Comment by chris-huggins** [comment]
@discodavey Once you're happy with this, I'm ok for it to be merged to Pattern Library. I'd like to to hold off deploying to journal for now though.

**Comment by chris-huggins** [comment]
@discodavey Same with this PR. Once you're happy with this I'm ok with it being merged to pattern library. But let's hold off deploying to journal for now. We could put both this and the MSAs into a staging environment when ready though.

**Comment by discodavey** [comment]
Just gone through latest changes with Cosmin. Just waiting for  @chris-huggins to approve

**Comment by chris-huggins** [comment]
This has also made teaser labels on article pages such as "Related to", "Builds upon" semibold, which was not intended. However, for the sake of time I'm ok with deploying this as it is, and we can decide if we need to resolve that later.

**Comment by Cosmin-Daniel** [comment]
> This has also made teaser labels on article pages such as "Related to", "Builds upon" semibold, which was not my intention. However, for the sake of time I'm ok with deploying this as it is, and we can decide if we need to resolve that later.

The labels on the related teasers, like "Related to", "Builds upon", have now normal font weight.

**Comment by Cosmin-Daniel** [comment]
@chris-huggins I've applied the new styling to the MSA labels in the hero-banner too.

File: assets/sass/patterns/organisms/hero-banner.scss [hunk 12]
```diff
@@ -124,7 +124,7 @@
     text-decoration: none;
 
     &:hover {
-      color: $color-primary-dark;
+      @include color-link-hover();
```
Comment (discodavey) [inline assets/sass/patterns/organisms/hero-banner.scss:127]: @chris-huggins Are we having underline on hover or color change for MSA's?

**Review by discodavey — COMMENTED** [review]
Dependent on @chris-huggins answer to the underline on hover or colour. This looks ok.

File: assets/sass/patterns/organisms/hero-banner.scss (hunk 12 above)
Comment (chris-huggins) [reply assets/sass/patterns/organisms/hero-banner.scss:127]: I like the underline. I think we should also have a slightly darker blue on hover. Is that the `color-link-hover` you've suggested? I can't see a colour change on hover at the moment.

**Review by nlisgo — APPROVED** [review]

**Comment by nlisgo** [comment]
@chris-huggins I am going to merge this and expect this to go live today. Can you give a thumbs up?

We can deal with any issues in a followup. Thanks for the work everyone!

**Comment by chris-huggins** [comment]
@nlisgo to staging or are you deploying to journal?

**Comment by nlisgo** [comment]
> @nlisgo to staging or are you deploying to journal?

I intend to deploy to production. Or would you prefer to preview in staging first? If yes, my recommendation is that we merge it in and then pull in latest code to feature branch of journal. Launch a staging site and then deal with an issues as an immediate followup to pattern-library. Is that ok?
